first_hour_on_vwb/working_with_resources.ipynb

```diff
@@ -855,58 +855,38 @@
     "        self.input_name = wu.TextInputWidget(\"<NAME>\", \"Name:\").get()\n",
     "        self.output = widgets.Output()\n",
     "        self.button = wu.StyledButton('Create dataset','Click to create a BigQuery dataset as a controlled resource.','plus').get()\n",
-    "        self.vb = widgets.VBox([\n",
-    "            self.label,\n",
-    "            self.warning,\n",
-    "            self.access_drop_down,\n",
-    "            self.cloning_drop_down,\n",
-    "            self.input_dataset_id,\n",
-    "            self.input_description,\n",
-    "            self.input_name,\n",
-    "            self.button,\n",
-    "            self.output\n",
-    "        ], layout=wu.vbox_layout)\n",
+    "        self.toggle_optional = widgets.Checkbox(False, description = \"Show optional inputs\",indent=False)\n",
+    "        self.toggle_optional.observe(self.toggle)\n",
+    "        self.vb = widgets.VBox(\n",
+    "            children = [self.toggle_optional,self.label,self.warning,self.access_drop_down,self.cloning_drop_down,self.input_description,self.input_name,self.button,self.output], \n",
+    "            layout=wu.vbox_layout\n",
+    "        )\n",
     "        self.button.on_click(self.create_bq_dataset)\n",
     "\n",
+    "    def toggle(self,event):\n",
+    "        if self.toggle_optional.value == True:\n",
+    "            self.vb.children = [self.toggle_optional,self.label,self.warning,self.access_drop_down,self.cloning_drop_down,self.input_dataset_id,self.input_description,self.input_name,self.button,self.output]\n",
```

Comment:
The long lists in this class are _mostly_ repeated. Could we declare a long list of common children once, and then modify as needed? Something like...

```
self.common_children = [self.toggle_optional, self.label, self.warning, self.access_drop_down ...]

self.vb.children = self.common_children.insert(6, self.input_dataset_id)
```

Repeat this comment for the other related sections.

Reply:
fixed!